### PR TITLE
Allow simple container literals as default values

### PR DESCRIPTION
**Pull request by JonathanPlasse**
- Implement PyCQA/flake8-pyi#358

### CODE REVIEW

File: crates/ruff/resources/test/fixtures/flake8_pyi/PYI011.pyi [hunk 1]
```diff
@@ -13,7 +13,7 @@ def f11(*, x: str = "x") -> None: ...  # OK
 def f13(
     x: list[
         str
-    ] = [  # Error PYI011 Only simple default values allowed for typed arguments
+    ] = [  # OK
```
Comment (charliermarsh) [inline crates/ruff/resources/test/fixtures/flake8_pyi/PYI011.pyi:16]: It looks like nested containers are disallowed -- is that right? And do we have any tests for those?

Comment (JonathanPlasse) [reply crates/ruff/resources/test/fixtures/flake8_pyi/PYI011.pyi:16]: - Yes, I will add the added tests from PyCQA/flake8-pyi#358.

**Comment by JonathanPlasse** [comment]
> typeshed (+0, -244)

Nice to see.

**Comment by charliermarsh** [comment]
Thank you!!

**Pull request merged**